src/Changes.php

```diff
@@ -0,0 +1,46 @@
+<?php
+declare(strict_types=1);
+
+namespace Roave\ApiCompare;
+
+use ArrayIterator;
+use Assert\Assert;
+use IteratorAggregate;
+
+final class Changes implements IteratorAggregate
+{
+    /** @var Change[] */
+    private $changes = [];
+
+    private function __construct()
+    {
+    }
+
+    public static function new(): self
```

Comment:
Reinventing the wheel i see

Reply:
I agree. It does not capture any meaningful value here.

Reply:
There are two ways to construct this:

 * `::new()`
 * `::fromArray()`

Fully qualifies having static named constructors here to avoid potential BC breaks in the future

Reply:
Hmm, That is a valid point and I guess it also follows the "conventions" of the project.